langchain/src/document_loaders/fs/multi_file.ts

```diff
@@ -0,0 +1,113 @@
+import type { extname as ExtnameT, resolve as ResolveT } from "node:path";
+import type { stat as StatT } from "node:fs/promises";
+import { Document } from "@langchain/core/documents";
+import { getEnv } from "@langchain/core/utils/env";
+import { BaseDocumentLoader } from "../base.js";
+import { type LoadersMapping, UnknownHandling } from "./directory.js";
+
+/**
+ * A document loader that loads documents from multiple files. It extends the
+ * `BaseDocumentLoader` class and implements the `load()` method.
+ * @example
+ * ```typescript
+ *
+ * const multiFileLoader = new MultiFileLoader(
+ *   ["path/to/file1.pdf", "path/to/file2.txt"],
+ *   {
+ *     ".pdf": (path: string) => new PDFLoader(path),
+ *   },
+ * );
+ *
+ * const docs = await multiFileLoader.load();
+ * console.log({ docs });
+ *
+ * ```
+ */
+export class MultiFileLoader extends BaseDocumentLoader {
```

Comment:
This implementation shares a lot in common with `DirectoryLoader` that I wonder if this is something that can be moved to a new base class so there is reuse.